### PR TITLE
provide details for debugging failing test-thredds notebook

### PR DESCRIPTION
**Pull request by fmigneault**
# Overview

Adds general debugging utilities regarding notebook `test_thredds`

## Changes

- Adds variable overrides in `test_thredds` for local debugging. 
- Adds important documentation notes regarding possible error debugging leads for `test_thredds` (based on actual issues encountered). 

## Related Issue / Discussion

- Relates to https://github.com/bird-house/birdhouse-deploy/pull/188#issuecomment-895622277

### CODE REVIEW

**Review by tlvu — APPROVED** [review]
LGTM. As usual, copy/paste PR description into merge commit description. Thx.

File: notebooks-auth/test_thredds.ipynb [hunk 1]
```diff
@@ -56,15 +56,15 @@
     "import requests\n",
     "print(\"Setup configuration parameters...\")\n",
     "\n",
-    "PAVICS_HOST = os.getenv(\"PAVICS_HOST\", \"pavics.ouranos.ca\").rstrip(\"/\")\n",
+    "PAVICS_HOST = (os.getenv(\"PAVICS_HOST\") or \"pavics.ouranos.ca\").rstrip(\"/\")\n",
```
Comment (tlvu) [inline notebooks-auth/test_thredds.ipynb:59]: Just curious, `os.getenv("VAR_NAME", default_value)` do not work so you have to do `os.getenv("VAR_NAME") or default_value`?  I know both achieve the same effect so I am not against the other format but I just wonder what is the subtleties that make you change to the other form.

Comment (fmigneault) [reply notebooks-auth/test_thredds.ipynb:59]: `os.getenv("VAR_NAME", default_value)` will return an empty string or None if the variable is defined with `VAR_NAME=` or `VAR_NAME=""`.  In this case, I want to consider empty equivalent as undefined and revert to the default.

Comment (tlvu) [reply notebooks-auth/test_thredds.ipynb:59]: I see!  That "empty value" problem did happen via Jenkins build request or you are being defensive for when invoking `pytest` manually directly on the command-line and user can accidentally introduce the problem?

**Pull request merged**